File: Data Structures & Algorithms/remove-element/submission-0.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        p1= 0
        p2=0

        for i, x in enumerate(nums):
            if x==val:
                continue
            else:
                nums[p1]=nums[i]
                p1+=1
                
        return p1

File: Data Structures & Algorithms/subsets/submission-0.cpp
class Solution {
public:
    void backtrack(vector<int>& nums, int start, vector<int> arr, vector<vector<int>>& res){
        if(start>=nums.size()){
            res.push_back(arr);
            return;
        }
        arr.push_back(nums[start]);
        backtrack(nums, start+1, arr, res);

        arr.pop_back();
        backtrack(nums, start+1, arr, res);
}

    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> res={};
        vector<int> arr={};
        int start=0;

        backtrack(nums, 0, arr, res);

        return res;



    }
};
